### PR TITLE
Added gene annotation tests and a test set for control

### PR DESCRIPTION
**Pull request by Fl-Sch**
Added gene annotation tests in file scr/memote.jl and a test set for control in test/memote.jl

### CODE REVIEW

**Review by exaexa — COMMENTED** [review]
Hi! This looks mostly okay, thanks for the contributions!

Few fixes required:
- please convert the comments to docstrings (see code above for templates)
- you might want to adjust your VSCode formatting settings (but don't care about formatting now, I can fix it quickly later)
- you might need to export your functions so that the test code can find them (see PR #15 comments for a similar problem)

File: src/memote.jl [hunk 1]
```diff
@@ -321,3 +321,59 @@ function is_consistent(
...
+    annos = gene_annotations(model, gid)
+    for anno_kw in keys(GENE_PRODUCT_ANNOTATIONS)
+      if haskey(annos, anno_kw) == true 
+        !in(anno_kw, keys(no_annos)) && (no_annos[anno_kw] = [])
+        in.(nothing, Ref(match.(GENE_PRODUCT_ANNOTATIONS[anno_kw], annos[anno_kw]))) && push!(no_annos[anno_kw], gid)
+      end  
+    end
+  end
+  return no_annos
+end
```
Comment (stelmo) [inline src/memote.jl:379]: Change the vscode code settings to add a new line at the end

**Review by stelmo — CHANGES_REQUESTED** [review]
Nice progress :) Just implement all the changes and re-request review

File: src/memote.jl [hunk 2]
```diff
@@ -321,3 +321,59 @@ function is_consistent(
...
+    "ecogene" => r"^EG\d+$", 
+    "kegg.genes" => r"^\w+:[\w\d\.-]*$", 
+    "ncbigi" => r"^(GI|gi)\:\d+$", 
+    "ncbigene" => r"^\d+$", 
+    "ncbiprotein" => r"^(\w+\d+(\.\d+)?)|(NP_\d+)$", 
+    "ccds" => r"^CCDS\d+\.\d+$", 
+    "hprd" => r"^\d+$",
+    "asap" => r"^[A-Za-z0-9-]+$")
+    
+  )
```
[the remark this answers is not in the document]
Comment (stelmo) [reply src/memote.jl:367]: Actually, on second thought, that is too much. Just keep it as is :) I will convert it at a later stage.

**Comment by exaexa** [comment]
@Fl-Sch when #20 is merged pls rebase this and add yourself to the list of authors :]

**Comment by stelmo** [comment]
you will have to rebase this PR since the structure of the code dramatically changed...

**Comment by Fl-Sch** [comment]
Should I add a new file gene.jl for the gene annotation tests in the memote folder like for metabolites and reactions?

**Comment by exaexa** [comment]
> Should I add a new file gene.jl for the gene annotation tests in the memote folder like for metabolites and reactions?

Yeah, technically that would make sense (esp. if there's gonna be more tests of that kind)

**Comment by Fl-Sch** [comment]
I have not fixed the formating  because I do not know how since I use windows.

**Comment by Fl-Sch** [comment]
Did you just format or did my /format work somehow?

**Comment by stelmo** [comment]
> Did you just format or did my /format work somehow?

Hmm I am not sure who formatted your PR. But if you comment `/format` the bot will format your PR for you. Alternatively, you can use JuliaFormatter (a package) to format your PR. Either way is fine, I usually just use `/format` but you need to `git fetch` before making more edits otherwise you will run into merge issues as your local branch will be out of sync with the remote branch.

File: src/memote/consistency.jl [hunk 3]
```diff
@@ -296,6 +296,7 @@ function model_is_consistent(model, optimizer; config = memote_config)
     termination_status(opt_model) == OPTIMAL
 end
 
+<< << << < HEAD:src/memote/consistency.jl
```
Comment (stelmo) [inline src/memote/consistency.jl:299]: what is up with this?

Comment (Fl-Sch) [reply src/memote/consistency.jl:299]: Ah sorry, git automatically wanted to add the gene annotation tests from the previous version of this PR to the consistency.jl file when I rebased for some reason. I moved them to the correct file but accidentally missed that line, I'll remove it now.

**Review by stelmo — APPROVED** [review]
looks good to me

**Review by exaexa — APPROVED** [review]
lgtm fftm

**Comment by stelmo** [comment]
weirdly some tests failed...

**Comment by exaexa** [comment]
`all_unannotated_genes` isn't exported it seems

**Comment by stelmo** [comment]
I will fix

**Comment by Fl-Sch** [comment]
Not sure if I can look over it today but I will look at it tomorrow

**Comment by exaexa** [comment]
EDIT: actually all 3 functions are not exported but only `all_unnanotated_genes` is displayed as the testset fails at that point.

Possible fixes:
- export all 3 functions (in `src/FBCModelTests.jl`)
- in tests, use the prefix, as in `FBCModelTests.all_unnannotated_genes` for all functions

**Comment by stelmo** [comment]
if you rebase on the current master then you won't have to do this as the auto export has been implemented there (see `FBCModelTests.jl`

**Comment by Fl-Sch** [comment]
Hope the problem is fixed now

**Comment by exaexa** [comment]
I also needed to push the file reference to the doc generator; now I guess all is okay. Thanks a lot!

**Pull request merged**